aws_xray_sdk/ext/pynamodb/patch.py

```diff
@@ -0,0 +1,61 @@
+import botocore.vendored.requests.sessions
+import json
+import wrapt
+
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.models import http
+from aws_xray_sdk.ext.boto_utils import _extract_whitelisted_params
+
+
+def patch():
+    """Patch PynamoDB so it generates subsegements when calling DynamoDB."""
+    if hasattr(botocore.vendored.requests.sessions, '_xray_enabled'):
+        return
+    setattr(botocore.vendored.requests.sessions, '_xray_enabled', True)
+
+    wrapt.wrap_function_wrapper(
+        'botocore.vendored.requests.sessions',
+        'Session.send',
+        _xray_traced_pynamodb,
+    )
+
+
+def _xray_traced_pynamodb(wrapped, instance, args, kwargs):
+
+    # Check if it's a request to DynamoDB and return otherwise.
+    try:
+        service = args[0].headers['X-Amz-Target'].decode('utf-8').split('_')[0]
+    except KeyError:
+        return
+    if service.lower() != 'dynamodb':
+        return
+
+    return xray_recorder.record_subsegment(
+        wrapped, instance, args, kwargs,
+        name='dynamodb',
+        namespace='aws',
+        meta_processor=pynamodb_meta_processor,
+    )
+
+
+def pynamodb_meta_processor(wrapped, instance, args, kwargs, return_value,
+                            exception, subsegment, stack):
+    operation_name = args[0].headers['X-Amz-Target'].decode('utf-8').split('.')[1]
```

Comment:
Is it possible to capture `request_id` here?

Reply:
Thanks for pointing out. Turned out I searched for it in the request instead of the response. Added it now.